pkg/gofr/handler.go

```diff
@@ -109,6 +109,14 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		err = gofrHTTP.ErrorPanicRecovery{}
 	}
 
+	// **Handle Custom Headers** if `result` is a `Response`.
+	if resp, ok := result.(response.Response); ok {
+		for key, value := range resp.Headers {
+			w.Header().Set(key, value)
```

Comment:
Can we add logic to skip headers that gofr already sets, such as 'Content-type'?

Reply:
`if resp, ok := result.(response.Response); ok {

	for key, value := range resp.Headers {
		if w.Header().Get(key) == "" {
			w.Header().Set(key, value)
		}
	}
	result = resp.Data
}`
How about this, we'll check if a Header has been set and then set it only if it hasn't been set.

Reply:
Yes maybe this can help

Reply:
LGTM.

Reply:
@Umang01-hash @aryan-mehrotra-zs @vipul-rawat How about having a list of non overridable response headers (or response headers that can only be overridden by some special methods provided by Gofr)? So that a check can be implemented for those headers only. Intention behind is that, user deliberately wants to override those headers (such as content-type) but since Gofr is internally setting those headers, so the overall working of the framework should not break.

**PS:** If required, this can be done as future enhancement. For now, this PR is good.

Reply:
I would suggest if the user has provided some header which is already set by gofr that should also be over-ridden, if user doesn't set - the default which is set by gofr is used.